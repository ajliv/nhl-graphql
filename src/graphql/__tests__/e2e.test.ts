import { gql } from 'apollo-server-express';

import { startTestServer } from './utils';

const FETCH_TEAM = gql`
    query TeamQuery($team: Int) {
        team(id: $team) {
            name
            abbreviation
            roster {
                primaryNumber
            }
        }
    }
`;

const FETCH_PLAYER = gql`
    query PlayerQuery($player: Int) {
        player(id: $player) {
            fullName
            age
            birthDate
            position
            currentTeam {
                name
            }
        }
    }
`;

describe('E2E Server', () => {
    let graphql: (op: any) => Promise<any>;
    let stop: () => Promise<any>;

    beforeEach(async () => {
        const server = await startTestServer();
        graphql = server.graphql;
        stop = server.stop;
    });

    afterEach(async () => {
        await stop();
    });

    it('fetches a team', async () => {
        const resp = await graphql({
            query: FETCH_TEAM,
            variables: { team: 25 },
        });
        expect(resp).toMatchSnapshot();
    });

    it('fetches a person', async () => {
        const resp = await graphql({
            query: FETCH_PLAYER,
            variables: { person: 8474849 },
        });
        expect(resp).toMatchSnapshot();
    });
});
