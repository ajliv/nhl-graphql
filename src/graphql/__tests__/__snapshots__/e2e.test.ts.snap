// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E Server fetches a person 1`] = `
Object {
  "data": Object {
    "player": Object {
      "age": 29,
      "birthDate": "1989-11-21T00:00:00.000Z",
      "currentTeam": Object {
        "name": "Dallas Stars",
      },
      "fullName": "Antoine Roussel",
      "position": "LW",
    },
  },
}
`;

exports[`E2E Server fetches a team 1`] = `
Object {
  "data": Object {
    "team": Object {
      "abbreviation": "DAL",
      "name": "Dallas Stars",
      "roster": Array [
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
        Object {
          "primaryNumber": 26,
        },
      ],
    },
  },
}
`;
